source/pele/xy_model_online.h

```diff
@@ -0,0 +1,232 @@
+#ifndef _PELE_XY_MODEL_ONLINE_H
+#define _PELE_XY_MODEL_ONLINE_H
+
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+#include "pele/base_potential_online.h"
+
+namespace pele {
+
+/**
+ * Adjacency list graph implementation according to S. S. Skiena,
+ * Algorithm Design Manual 
+ */
+
+struct edgenode {
+    const unsigned int y;
+    edgenode* next;
+    edgenode(const unsigned int y_, edgenode*const& next_)
+        : y(y_),
+          next(next_)
+    {}
+};
+
+class graph {
```

Comment:
I agree. When looking at the existing graph I also had the impression it wanted to achieve something different. For division of the xy potential into batches adjacency lists are convenient. "adjacency_list_" is now added.